### PR TITLE
Revert PR that shows onboarding payments setup URL in authenticated webview

### PR DESCRIPTION
**Pull request by jaclync**
Reverts https://github.com/woocommerce/woocommerce-ios/pull/14252

## Description

As @joshheald found an issue when adding Stripe information in the payments setup webview from the payments onboarding flow due to the changes in https://github.com/woocommerce/woocommerce-ios/pull/14252 pdfdoF-5Nj-p2, this PR reverts the PR that causes it so that it does not affect production users. I will investigate deeper into this issue https://github.com/woocommerce/woocommerce-ios/issues/14303 after this beta fix, likely later this week.

## Steps to reproduce

Since this change affects the existing onboarding flow for all entry points (POS, IPP, Payments menu), I think focusing on testing the most critical flow, IPP, is the best use of time.

### Stripe overdue

Prerequisite: store payment gateway is active and set up but account has `account.wcpayStatus == .restricted && account.hasOverdueRequirements`. Alternatively, you can hard code the state by returning `.stripeAccountOverdueRequirement(plugin: .wcPay)` at the beginning of `CardPresentPaymentsOnboardingUseCase.checkOnboardingState`.

- Log in to the store with site credentials
- Go to the orders tab
- Create an order
- Enter a custom amount or add some products then recalculate
- Collect payment
- Tap `Card Reader` --> an onboarding modal should be shown. after the loading state, the Stripe overdue UI should be shown
- Tap `Resolve Now` --> the webview should be shown in a Safari sheet
- Log in to wp-admin if needed
- Proceed with the onboarding flow --> should be able to navigate to the Stripe information form
- Log out and log in with WPCOM
- Repeat the steps above

### Payments plugin not set up

Prerequisite: store payment gateway is active but isn't set up completely. Alternatively, you can hard code the state by returning `.pluginSetupNotCompleted(plugin: .wcPay)` at the beginning of `CardPresentPaymentsOnboardingUseCase.checkOnboardingState`.

- Log in to the store with site credentials
- Go to the orders tab
- Create an order with a custom amount or product, or tap on an order ready for payment collection
- Collect payment
- Tap `Card Reader` --> an onboarding modal should be shown. after the loading state, the plugin not set up UI should be shown
- Tap `Finish Setup in Store Admin` --> the webview should be shown in a Safari sheet
- Log in to wp-admin if needed
- Proceed with the onboarding flow --> should be able to navigate to the Stripe information form
- Log out and log in with WPCOM
- Repeat the steps above

## Testing information

I tested with a JN site on iPad Pro 11-inch 3rd gen, iOS 17.5.1.

## Screenshots

https://github.com/user-attachments/assets/44af2a16-87e2-4aab-99a7-f2485a1f57f2

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/230">21.0 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR contains changes to <code>RELEASE-NOTES.txt</code>.<br />
Note that these changes won't affect the final version of the release notes as this version is in code freeze.<br />
Please, get in touch with a release manager if you want to update the final release notes.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by jaclync** [comment]
Hi @rachelmcr 👋 this is a beta fix to revert a PR in 21.0 that introduces regressions in the existing payments onboarding flow (can be triggered by IPP, Payments menu, refund). It's a simple revert for all changes in the PR, and I will look into a proper fix separately. I will let you know when it's ready to merge.

**Comment by jaclync** [comment]
Note for reviewer: 2 unit tests are failing consistently (`StorePerformanceViewModelTests.test_dates_for_custom_range_are_correct_for_non_custom_time_range()`
and `TopPerformersDashboardViewModelTests.test_dates_for_custom_range_are_correct_for_non_custom_time_range()`) that were fixed in trunk after the release branch was created https://github.com/woocommerce/woocommerce-ios/pull/14287, I cherry-picked the merge commit to fix them in the PR. Hopefully, there will be no issues when the release branch is merged back into trunk.

File: RELEASE-NOTES.txt
Comment (joshheald) [inline RELEASE-NOTES.txt:11]: [The user-facing release notes are already taken](https://github.com/woocommerce/woocommerce-ios/blob/3087192c9da65e987e2688d45a3c0ba104a8192a/WooCommerce/Resources/release_notes.txt) and show:

> Product and order management just got easier! You can now view and edit custom fields for products and orders right within the app. Also, we've smoothed out the payments onboarding process. It's now easier than ever to resolve errors setting up WooPayments or Stripe. Happy selling!

I don't think we need to change them to remove that passing comment, even though we haven't made any changes to onboarding once this is reverted. Changing them would mean stopping the translation process and resubmitting, etc etc. Just thought it was worth noting though.

**Review by joshheald — APPROVED** [review]
Works as expected – I completed onboarding in the webview successfully.

File: RELEASE-NOTES.txt
Comment (jaclync) [reply RELEASE-NOTES.txt:11]: Good point, I feel like the release notes read well by just taking out the two sentences `Also, we've smoothed out the payments onboarding process. It's now easier than ever to resolve errors setting up WooPayments or Stripe.` for the English version. @rachelmcr do you think it's okay for me to update the [user-facing release notes](https://github.com/woocommerce/woocommerce-ios/blob/3087192c9da65e987e2688d45a3c0ba104a8192a/WooCommerce/Resources/release_notes.txt) in this PR? I'm not sure if it might interrupt the translation and submission process. I plan to merge the PR in an hour.

Comment (rachelmcr) [reply RELEASE-NOTES.txt:11]: > do you think it's okay for me to update the [user-facing release notes](https://github.com/woocommerce/woocommerce-ios/blob/3087192c9da65e987e2688d45a3c0ba104a8192a/WooCommerce/Resources/release_notes.txt) in this PR? I'm not sure if it might interrupt the translation and submission process.

I'm fairly sure changing it at this point will invalidate any existing translations, so we'd have to resubmit the text for translation (which might not get done in time). I think I'd err on the side of leaving the release notes as they are. It probably won't do any harm to have that mention in the release notes, whereas not having the release notes translated would be more disruptive.

Comment (jaclync) [reply RELEASE-NOTES.txt:11]: Makes sense! I just merged the PR, thanks in advance @rachelmcr for making the beta build 🙏

**Comment by rachelmcr** [comment]
Thanks for pinging me here! We have one other beta fix PR so I'll release a new beta once that's merged, too.

**Pull request merged**
